crates/router/src/connector/stripe.rs

```diff
@@ -1915,7 +1923,11 @@ impl api::IncomingWebhook for Stripe {
                         match meta_data.is_refund_id_as_reference {
                             // if the order_id is refund_id
                             Some(_) => api_models::webhooks::ObjectReferenceId::RefundId(
-                                api_models::webhooks::RefundIdType::RefundId(meta_data.order_id),
+                                api_models::webhooks::RefundIdType::RefundId(
```

Comment:
Same here, Instead of checking metadata, extract order_id and match order_id for some or None